scipy/interpolate/tests/test_interpnd.py

```diff
@@ -166,6 +171,49 @@ def test_pickle(self):
 
         assert_almost_equal(ip(0.5, 0.5), ip2(0.5, 0.5))
 
+    @pytest.mark.slow
+    @pytest.mark.skipif(_IS_32BIT, reason='it fails on 32-bit')
+    def test_threading(self):
+        # This test was taken from issue 8856
+        # https://github.com/scipy/scipy/issues/8856
+        check_free_memory(30000)
```

Comment:
Did you really mean to set the limit to 30 GB? That means it won't be run on machine with 32 GB RAM - i.e., almost never.

I seem to be needing about 10 GB total:
```
$ ulimit -Sv 10000000
$ python dev.py test -t scipy.interpolate.tests.test_interpnd::TestLinearNDInterpolation::test_threading -m full
```
It crashes with 8 GB on my machine. I guess that makes sense, since the 32-bit test showed that a single array creation needed about 1.5 GB, and there are four threads here. So the test needs 6 GB, and then there's the memory taken by SciPy itself, other tests, and dependencies.

Ideally I think we set the limit to 10 GB or so, and tweak the test a bit so that it requires 0.5 GB per thread. That way it'll run for most contributors.

Could you please check if the test still reproducibly crashes when you slightly lower the array sizes?
```

Reply:
Let me check it again with lower bounds

Reply:
With the new sizes, the reproducible case now consumes ~1Gb:

```
Command being timed: "python qhull_err.py"
User time (seconds): 53.63
System time (seconds): 83.40
Percent of CPU this job got: 551%
Elapsed (wall clock) time (h:mm:ss or m:ss): 0:24.85
Average shared text size (kbytes): 0
Average unshared data size (kbytes): 0
Average stack size (kbytes): 0
Average total size (kbytes): 0
Maximum resident set size (kbytes): 1128924
Average resident set size (kbytes): 0
Major (requiring I/O) page faults: 1
Minor (reclaiming a frame) page faults: 133340
Voluntary context switches: 2483779
Involuntary context switches: 841
Swaps: 0
File system inputs: 0
File system outputs: 8
Socket messages sent: 0
Socket messages received: 0
Signals delivered: 0
Page size (bytes): 4096
Exit status: 0
```

Reply:
Unfortunately, the lowest values for which we can reliably reproduce the crash case imply a memory consumption of ~5GB

Reply:
That seems fine - that'll mean it runs on most developer machines. Let's call this good.